Add db:migrate:status rake task to check which migrations will be run.

FabienChaynes opened the pull request:
This PR adds a `rake db:migrate:status` task listing pending migrations. It accepts an optionnal `VERSION` env variable which will be used as the targeted version (just like in `rake db:migrate`).

The output will look like this:
```
database: mongoid_test

 Status   Migration ID    Migration Name
--------------------------------------------------
   up     20100513054656  AddBaselineSurveySchema
   up     20100513055502  AddSecondSurveySchema
   up     20100513063902  AddImprovementPlanSurveySchema
```
This formating is inspired by [ActiveRecord rake db:migrate output](https://github.com/rails/rails/blob/master/activerecord/lib/active_record/tasks/database_tasks.rb#L205).

FabienChaynes commented:
Sure, done!

I did not bump the version number though because the [latest released version](https://rubygems.org/gems/mongoid_rails_migrations/) seems to be  `1.1.1` and it was already bumped to `1.2.0` but not released.

adacosta commented on lib/mongoid_rails_migrations/active_record_ext/migrations.rb:
```diff
@@ -198,6 +198,14 @@ def migrate(migrations_path, target_version = nil)
 
+      def status(migrations_path, target_version = nil)
+        case
+          when target_version.nil?              then self.new(:up, migrations_path, target_version).status
```
I suggest removing the "then" and moving the "then" action to the next line and also using new instead of self.new - they do the same thing.

adacosta commented on lib/mongoid_rails_migrations/active_record_ext/migrations.rb:
```diff
@@ -338,6 +334,27 @@ def migrate
     end
 
+    def status
+      runnable = runnable_migrations
```
Instead of creating this assignment, what about simply using it on line 340? Also, it looks like you could split the to_run variable in to a separate method to keep the status method more focused.

FabienChaynes commented:
I removed the useless `self` and extracted some logic from `#status` into another method.

I did not remove the `then` though because all the one line `when` in the gem are written like this so it'd be inconsistent. If you still want me to do it for this case (or for all the cases) just tell me and I'll change it :slightly_smiling_face:

jarthod commented:
Looks good, please add the command to the list in the readme too ;)

FabienChaynes commented:
Done.

jarthod commented:
Released in `1.2.0`
